### PR TITLE
[Fleet] Support download rate in string with unit format

### PR DESCRIPTION
**Pull request by nchaulet**
## Description 

Relates https://github.com/elastic/ingest-dev/issues/3446

Elastic-agent are currently not sending download rate as a float but as a string with unit, while this is fixed on the agent side and for older agent, Fleet server should accept both format.

That PR does that, and return a 400 instead of a 500 when an error parsing meta happen.

## Todo

- [x] Changelog

### CODE REVIEW

File: internal/pkg/api/metadata.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/docker/go-units"
+)
+
+func parseDownloadRate(jsonDownloadRate json.RawMessage) (*float64, error) {
```
Comment (michel-laterman) [inline internal/pkg/api/metadata.go:15]: We should adjust the openapi spec for the download rate to be:
```yaml
oneOf:
  - type: string
  - type: integer
```

**Review by michel-laterman — CHANGES_REQUESTED** [review]

File: internal/pkg/api/metadata.go (hunk 1 above)
Comment (nchaulet) [reply internal/pkg/api/metadata.go:15]: I am wondering if we should change the openAPISpec, as this mostly a temporary fix to support a client that did not sent the correct value.

Comment (cmacknz) [reply internal/pkg/api/metadata.go:15]: Don't make the error an officially accepted value. Tolerate it but don't encourage it's use.

Comment (nchaulet) [reply internal/pkg/api/metadata.go:15]: Yes I will add one with a random string input already had one with `MBps`

File: changelog/fragments/1720021515-fix_download_rate.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+# - breaking-change: a change to previously-documented behavior
+# - deprecation: functionality that is being removed in a later release
+# - bug-fix: fixes a problem in a previous version
+# - enhancement: extends functionality but does not break or fix existing behavior
+# - feature: new functionality
+# - known-issue: problems that we are aware of in a given version
+# - security: impacts on the security of a product or a user’s deployment.
+# - upgrade: important information for someone upgrading from a prior version
+# - other: does not fit into any of the other categories
+kind: feature
```
[the remark this answers is not in the document]
Comment (nchaulet) [reply changelog/fragments/1720021515-fix_download_rate.yaml]: bug fix 👍

**Comment by nchaulet** [comment]
@michel-laterman I think it will make sense to backport this 8.15 no?

File: internal/pkg/api/metadata_test.go [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+		name:          "download rate as float",
+		raw:           json.RawMessage(`1000000`),
+		expectedValue: 1000000.00,
+	}, {
+		name:          "download rate as MBps",
+		raw:           json.RawMessage(`"1MBps"`),
+		expectedValue: 1000000.00,
+	}, {
+		name:           "download rate random string",
+		raw:            json.RawMessage(`"toto"`),
```
Comment (michel-laterman) [inline internal/pkg/api/metadata_test.go:33]: My suggestion for an additional test case would be to include just the suffix that is checked:
```json.RawMessage(`"MBps"`)``` as a test case

**Review by michel-laterman — APPROVED** [review]
Please backport and add a an additional test case

**Pull request merged**